src/Controls/src/Core/Platform/GestureManager/GestureManager.Windows.cs

```diff
@@ -511,6 +519,81 @@ void OnPointerReleased(object sender, PointerRoutedEventArgs e)
 			PanComplete(true);
 		}
 
+		void OnPgrPointerEntered(object sender, PointerRoutedEventArgs e)
+		{
+			var view = Element as View;
+			if (view == null)
+				return;
+
+			var pointerPoint = e.GetCurrentPoint(Control);
+			var children = (view as IGestureController)?.GetChildElements(new Point(pointerPoint.Position.X, pointerPoint.Position.Y));
+
+			if (children != null)
+				foreach (var recognizer in children.GetChildGesturesFor<PointerGestureRecognizer>())
+				{
+					recognizer.SendPointerEntered(view);
+				}
+
+			if (e.Handled)
+				return;
+
+			IEnumerable<PointerGestureRecognizer> pointerGestures = view.GestureRecognizers.GetGesturesFor<PointerGestureRecognizer>();
+			foreach (var recognizer in pointerGestures)
+			{
+				recognizer.SendPointerEntered(view);
+			}
+		}
+
+		void OnPgrPointerExited(object sender, PointerRoutedEventArgs e)
```

Comment:
I think there is a lot of repeated code in the first parts of all of the added methods, right? Can we come up with a way to combine those into one and only triggering the right event?

Maybe a method that takes in a func that we trigger and we replace all the bodies of these methods with just 1 line?

Reply:
done! used Action since they were void. thanks @PureWeen for the help